protocols/request-response/tests/ping.rs

```diff
@@ -38,6 +38,30 @@ use rand::{self, Rng};
 use std::{io, iter};
 use std::{collections::HashSet, num::NonZeroU16};
 
+/// Check if is_response_outbound works properly
+#[test]
+fn pending_connections() {
+   let ping = Ping("ping".to_string().into_bytes());
+   let offline_peer = PeerId::random();
+
+   let protocols = iter::once((PingProtocol(), ProtocolSupport::Full));
+   let cfg = RequestResponseConfig::default();
+
+   let (peer1_id, trans) = mk_transport();
+   let ping_proto1 = RequestResponse::throttled(PingCodec(), protocols.clone(), cfg.clone());
+   let mut swarm1 = Swarm::new(trans, ping_proto1, peer1_id.clone());
+
+   let request_id_1 = swarm1.send_request(&offline_peer, ping.clone()).unwrap();
+
+   // Poll swarm until request 1 is reported as failed.
+   swarm1.network.poll();
```

Comment:
Polling the swarm a single time likely won't do it here.

In case you want to read up on the concept of asynchronous programming in Rust, or the concept of a `Future` and its `poll` function, I recommend the [async-book](https://rust-lang.github.io/async-book/01_getting_started/01_chapter.html).

Below is a concrete example of polling a swarm until some event X happens:

https://github.com/libp2p/rust-libp2p/blob/6d0773b68e3e2d820d28dfc3d2486bfbb80f3248/protocols/request-response/tests/ping.rs#L98-L116